encoding/thrift/thriftrw-plugin-yarpc/main.go

```diff
@@ -322,6 +314,10 @@ func (generator) Generate(req *api.GenerateServiceRequest) (*api.GenerateService
 	return &api.GenerateServiceResponse{Files: files}, nil
 }
 
+func splitFunctionPath(input string) (string, string) {
+	return input[0:strings.LastIndex(input, ".")], input[strings.LastIndex(input, ".")+1 : len(input)]
```

Comment:
You can omit 0 in `0:x` and `len(s)` in `x:len(s)` to make this,

    return input[:i], input[i+1:]